app/controllers/avo/associations_controller.rb

```diff
@@ -190,5 +194,42 @@ def has_many_reflection?
         ActiveRecord::Reflection::HasAndBelongsToManyReflection
       ]
     end
+
+    def through_reflection?
+      @reflection.instance_of? ActiveRecord::Reflection::ThroughReflection
+    end
+
+    def additional_params?
+      additional_params.keys.count >= 1
+    end
+
+    def additional_params
+      params[:fields].except("related_id")
+    end
+
+    def set_attach_fields
+      @attach_fields = if @field.attach_fields.present?
+        Avo::FieldsExecutionContext.new(target: @field.attach_fields)
+          .detect_fields
+          .items_holder
+          .items
+      end
+    end
+
+    def new_join_record
+      field_names = @attach_fields.map { |field| field.name.tr(" ", "_").downcase }
+
+      @resource.fill_record(
```

Comment:
@adrianthedev We use `fill_record` which permits and assigns only those values whose key match the fields declared by the user in their resources file. I think this mitigates the chances of such attacks, no?

https://github.com/avo-hq/avo/blob/90a64f938798049e4749b498400a0c0091623508/lib/avo/resources/base.rb#L448

Reply:
> I think this mitigates the chances of such attacks, no?

Yes, but let's apply the `permit` directly inside the `additional_params` method to increase readability.

```ruby
def additional_params
  params[:fields].permit(@attach_fields.map { |field| field.name.tr(" ", "_").downcase })
end
```

Reply:
okayy. Got it :smile: